File: core/evidence/unsupported_continuity_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def _continuation_record(reason: str, evidence_gap: Dict[str, Any]) -> Dict[str, Any]:
    return {
        "reason": reason,
        "boundary_violation": {"type": "unsupported_continuity"},
        "evidence_gap": evidence_gap,
        "semantic_instability": {"unstable": True},
        "fragility": {"level": 0.7},
        "confidence_caps": {"max": 0.4},
    }


def suppress_unsupported_continuity(
    label: str,
    evidence: List[str],
    *,
    min_evidence: int = 2,
) -> Dict[str, Any]:
    gap = {"required": min_evidence, "actual": len(evidence)}
    if len(evidence) < min_evidence:
        return {"suppressed": True, "record": _continuation_record(f"continuity:{label}", gap)}
    return {"suppressed": False, "record": None}


def collect_unsupported_continuity(
    evidence: List[str],
    inferred: Dict[str, Any],
    reconciled: Dict[str, Any],
) -> List[Dict[str, Any]]:
    out: List[Dict[str, Any]] = []
    for k in inferred:
        r = suppress_unsupported_continuity(f"infer:{k}", evidence)
        if r["suppressed"] and r["record"]:
            out.append(r["record"])
    if reconciled and len(evidence) < 2:
        r = suppress_unsupported_continuity("reconcile", evidence)
        if r["suppressed"] and r["record"]:
            out.append(r["record"])
    return out

File: src/evidence/unsupportedContinuityEngine.ts
/**
 * Converted from Python: core/evidence/unsupported_continuity_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function _continuationRecord(reason: any, evidence_gap: any): any {
  return {"reason": reason, "boundary_violation": {"type": "unsupported_continuity"}, "evidence_gap": evidence_gap, "semantic_instability": {"unstable": true}, "fragility": {"level": py.F(0.7)}, "confidence_caps": {"max": py.F(0.4)}};
}
export function suppressUnsupportedContinuity(label: any, evidence: any, min_evidence: any = 2): any {
  var gap: any = {"required": min_evidence, "actual": py.len(evidence)};
  if ((py.len(evidence) < min_evidence)) {
    return {"suppressed": true, "record": _continuationRecord(`continuity:${py.toStr(label)}`, gap)};
  }
  return {"suppressed": false, "record": null};
}
export function collectUnsupportedContinuity(evidence: any, inferred: any, reconciled: any): any {
  var out: any[] = [];
  var k: any;
  for (k of py.iter(inferred)) {
    var r: any = suppressUnsupportedContinuity(`infer:${py.toStr(k)}`, evidence);
    if ((py.truthy(py.at(r, "suppressed")) && py.truthy(py.at(r, "record")))) {
      py.listAppend(out, py.at(r, "record"));
    }
  }
  if ((py.truthy(reconciled) && (py.len(evidence) < 2))) {
    r = suppressUnsupportedContinuity("reconcile", evidence);
    if ((py.truthy(py.at(r, "suppressed")) && py.truthy(py.at(r, "record")))) {
      py.listAppend(out, py.at(r, "record"));
    }
  }
  return out;
}
